polar/src/polar.lalrpop

```diff
@@ -202,207 +175,228 @@ RewritableOperator: Operator = {
     "in" => Operator::In,
 };
 
-pub RewrittenOperation: Operation = {
+RewrittenOperation: Value = {
     <op:RewritableOperator> "(" <mut args:(<TermExp> ",")*> <arg:TermExp?> ")" => {
         match arg {
             Some(arg) => args.push(arg),
             None => ()
         };
-        Operation{operator: op, args: args}
+        Value::Expression(Operation{operator: op, args: args})
     },
 };
 
-pub Predicate: Predicate = {
-    <name:Name> "(" <mut args:(<TermExp> ",")*> <arg:TermExp?> ")" => {
-        match arg {
-            Some(arg) => args.push(arg),
-            None => ()
-        };
-        Predicate{name, args}
-    }
-};
 
-Exp11<T>: Term = {
-    <term:Term> if T == "Term" => term,
-    <term:Pattern> if T == "Pattern" => term,
-    "(" <exp:Exp<T>> ")" => exp // seems sorta impossible to support parrens for expressions and list literals.
-}
+New: Value = "new" <literal:Spanned<InstanceLiteralTerm>> => {
+    let args = vec![literal];
+    let op = Operation{operator: Operator::New, args};
+    Value::Expression(op)
+};
 
-Exp10<T>: Term = {
+SpecialExp<T>: Value = {
```

Comment:
Why is this a special exp? I think the naming based on precedence (Exp10) is clearer.

Reply:
This one is not ideal because there isn't really a common thread for why these are grouped.

What I like about this pattern: the name of the parser rule matches the thing it is parsing. So we have AddExp, MatchExp, MulExp - mostly all descriptive names. The naming of levels is still there, e.g.

```
Exp10<T>: Term = {
    <Spanned<SpecialExp<T>>>,
    <Exp11<T>>,
}
```

Part of this is a side effect from wanting to refactor out all the term creations. You either have expressions are values, and the term wrapping is done in the rule, e.g. `AddOp: Value = <Spanned<Exp1>> "+" <Spanned<Exp2>>` or whatever,
Or all Exps are terms (so you don't need the Spanned) and you need to somehow wrap `AddOp` in `Spanned` (which is what we have here).

Reply:
Keyword Operation? `new` & `cut` are keywords.  `forall` could be moved to be with debug & print possibly.

Reply:
Moved them into BuiltinOperators and made them part of `Value` - they don't actually need to be parsed as expressions since they work as standalone values.